integrations/oci/oci-secrets-config-source/src/main/java/io/helidon/integrations/oci/secrets/configsource/OciSecretsConfigSourceProvider.java

```diff
@@ -0,0 +1,526 @@
+        @Override // NodeConfigSource
+        public Optional<NodeContent> load() {
+            return this.loader.get();
+        }
+
+        @Deprecated // For use by the Helidon Config subsystem only.
+        @Override // PollableSource
+        public Optional<PollingStrategy> pollingStrategy() {
+            return super.pollingStrategy();
+        }
+
+        private Optional<NodeContent> absentNodeContent() {
+            this.mostDistantExpirationInstant = null; // volatile write
+            return ABSENT_NODE_CONTENT;
+        }
+
+
+        /*
+         * Static methods.
+         */
+
+
+        private static Builder builder() {
+            return new Builder();
+        }
+
+        private static void completeAllSecretsTasks(Collection<? extends Callable<Void>> tasks, AutoCloseable secrets) {
+            RuntimeException re = null;
+            try (ExecutorService es = newVirtualThreadPerTaskExecutor()) {
+                for (Future<?> future : es.invokeAll(tasks)) {
+                    try {
+                        future.get();
+                    } catch (RuntimeException e) {
+                        if (re == null) {
+                            re = e;
+                        } else {
+                            re.addSuppressed(e);
+                        }
+                    } catch (Exception e) {
+                        if (e instanceof InterruptedException) {
+                            Thread.currentThread().interrupt();
+                        }
+                        if (re == null) {
+                            re = new IllegalStateException(e.getMessage(), e);
+                        } else {
+                            re.addSuppressed(e);
+                        }
+                    }
+                }
+            } catch (RuntimeException e) {
+                re = e;
+            } catch (Exception e) {
+                if (e instanceof InterruptedException) {
+                    Thread.currentThread().interrupt();
+                }
+                re = new IllegalStateException(e.getMessage(), e);
+            } finally {
+                try {
+                    secrets.close();
+                } catch (RuntimeException e) {
+                    if (re == null) {
+                        throw e;
+                    } else {
+                        re.addSuppressed(e);
+                    }
+                } catch (Exception e) {
+                    if (e instanceof InterruptedException) {
+                        Thread.currentThread().interrupt();
+                    }
+                    if (re == null) {
+                        throw new IllegalStateException(e.getMessage(), e);
+                    } else {
+                        re.addSuppressed(e);
+                    }
+                }
+                if (re != null) {
+                    throw re;
+                }
+            }
+        }
+
+        @SuppressWarnings("try")
```

Comment:
Why is this suppress warning present?

Reply:
Unless I put it in the wrong place, it is because OCI clients are `AutoCloseable` and do not override `close()`. This means that `close()` can throw `Exception`. Warnings are emitted from the compiler in such a case that something in a try-with-resources block can throw an `InterruptedException`. They are annoying and I am handling the `InterruptedException` case so I felt they were appropriate. What is your suggested action?